### PR TITLE
Fix `selector-{type,pseudo-class}-no-unknown` and `selector-type-case` false positives for deprecated selectors

### PR DESCRIPTION
**Pull request by Mouvedia**
> Which issue, if any, is this issue related to?

#8225

> Resources

- pseudo-classes
  - [`:top-layer`](https://issues.chromium.org/issues/40860536)
  - [`:popup-open`](https://issues.chromium.org/issues/40218886#comment39)
  - [`:focus-ring`](https://web.archive.org/web/20171209101655/https://drafts.csswg.org/selectors-4/)
  - [`:matches`](https://caniuse.com/css-matches-pseudo)
  - [`:drop`](https://drafts.csswg.org/selectors-4/#changes-2018-02)
    - [GTK](https://github.com/GNOME/gtk/commit/158dbbc88f2c2735c417d27bbc96dbecd7a67656) 
  - [`:user-error`](https://drafts.csswg.org/selectors-4/#changes-2013)
  - [`:contains`](https://www.w3.org/TR/2001/CR-css3-selectors-20011113/#content-selectors)
    - [sizzle](https://github.com/jquery/sizzle/wiki#additions)
    - [Beautiful Soup](https://facelessuser.github.io/soupsieve/selectors/pseudo-classes/#:-soup-contains)
- SVG elements
  - [Scripting and Interactivity chapter](https://svgwg.org/svg2-draft/single-page.html#changes-interact)
  - [Text chapter](https://svgwg.org/svg2-draft/single-page.html#changes-text)
  - [Fonts chapter](https://svgwg.org/svg2-draft/single-page.html#changes-fonts)
- `shadow` element
  - [biome](https://github.com/biomejs/biome/blob/main/crates/biome_css_analyze/src/keywords.rs#L5581)
  - [bugzilla.mozilla.org](https://bugzilla.mozilla.org/show_bug.cgi?id=887538)
  - [developer.mozilla.org](https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Releases/29)
  - [w3c.github.io](https://web.archive.org/web/20150429040731/http://w3c.github.io/webcomponents/spec/shadow/#the-shadow-element)

> Is there anything in the PR that needs further explanation?

Since Mozilla never went ahead with https://bugzilla.mozilla.org/show_bug.cgi?id=371323 back in 2007 I chose not to include `:nth-column` and `:nth-last-column`. They were supported by [`sel`](https://github.com/amccollum/sel) though.
We can discuss whether we should add these 2 and `:active-drop-target`, `:valid-drop-target`, `:invalid-drop-target`, `:nth-match` and `:nth-last-match` in #8225.
i.e. this PR goal is not to be exhaustive

### CODE REVIEW

File: lib/reference/selectors.mjs
Comment (Mouvedia) [inline lib/reference/selectors.mjs:1]: The 3 news sets are exported for #8225.

File: lib/rules/selector-type-no-unknown/index.mjs [hunk 1]
```diff
@@ -100,11 +100,12 @@ const rule = (primary, secondaryOptions) => {
 
 				const tagName = tagNode.value;
 				const tagNameLowerCase = tagName.toLowerCase();
+				// SVG tags are case-sensitive
+				const svgTypeSelectors = [...svgTags, 'hatch', 'hatchpath', 'hatchPath'];
```
Comment (Mouvedia) [inline lib/rules/selector-type-no-unknown/index.mjs:104]: see https://www.w3.org/TR/SVG/changes.html#pservers

File: lib/reference/selectors.mjs [hunk 2]
```diff
@@ -231,6 +255,15 @@ export const linguisticPseudoClasses = new Set(['dir', 'lang']);
 
 export const logicalCombinationsPseudoClasses = new Set(['has', 'is', 'matches', 'not', 'where']);
 
+export const deprecatedPseudoClasses = new Set([
+	'popup-open',
+	'top-layer',
+	'focus-ring',
+	'user-error',
+	'matches',
```
Comment (Mouvedia) [inline lib/reference/selectors.mjs:263]: note

`:matches` is already part of `logicalCombinationsPseudoClasses `

File: lib/reference/selectors.mjs [hunk 3]
```diff
@@ -85,6 +106,7 @@ export const mixedCaseSvgTypeSelectors = new Set([
 	'feTurbulence',
 	'foreignObject',
 	'glyphRef',
+	'hatchPath',
```
Comment (Mouvedia) [inline lib/reference/selectors.mjs:109]: > [Renamed 'hatchPath' to 'hatchpath'](https://www.w3.org/TR/SVG/changes.html#pservers)

File: lib/rules/selector-type-no-unknown/index.mjs (hunk 1 above)
Comment (Mouvedia) [inline lib/rules/selector-type-no-unknown/index.mjs:104]: cf https://github.com/element-io/svg-tags/blob/master/lib/svg-tags.json

File: lib/reference/selectors.mjs [hunk 4]
```diff
@@ -231,6 +255,15 @@ export const linguisticPseudoClasses = new Set(['dir', 'lang']);
 
 export const logicalCombinationsPseudoClasses = new Set(['has', 'is', 'matches', 'not', 'where']);
 
+export const deprecatedPseudoClasses = new Set([
```
Comment (Mouvedia) [inline lib/reference/selectors.mjs:258]: note

Depending on whether the PR that adds `selector-no-deprecated` will have a fixer or not, `deprecatedPseudoClasses` should or should not be exported.

File: lib/reference/selectors.mjs [hunk 5]
```diff
@@ -231,6 +255,15 @@ export const linguisticPseudoClasses = new Set(['dir', 'lang']);
...
+	'popup-open',
+	'top-layer',
+	'focus-ring',
+	'user-error',
+	'matches',
+	'contains',
+]);
```
Comment (Mouvedia) [inline lib/reference/selectors.mjs:265]:
```suggestion
export const deprecatedPseudoClasses = new Set([
	'contains',
	'drop',
	'focus-ring',
	'matches',
	'popup-open',
	'top-layer',
	'user-error',
]);
```

https://www.w3.org/TR/2022/WD-selectors-4-20220507/#changes-2018-02

**Review by jeddy3 — APPROVED** [review]
LGTM, thank you.

And thank you for diving into the specs to uncover these.

**Pull request merged**